### DISCUSSION TITLE
Families with largest size differentials

### DISCUSSION THREAD

**Discussion by Xiaojun928** [hahnlab/CAFE5#142] [Q&A] [answered]
Hello,

I'm using CAFE to estimate the gene gain and loss event. I've installed it and it works fine for the example data. However, I encountered the problem when running my own data. Here is the log file:

> Command line: cafexp -i iqtree.txt -t iqtree.tre 
> 
> Filtering families not present at the root from: 5290 to 2929
> 
> No root family size distribution specified, using uniform distribution
> 
> Optimizer strategy: Nelder-Mead with similarity cutoff
> Iterations: 300
> Expansion: 2
> Reflection: 1
> 
> Starting Search for Initial Parameter Values
> Lambda: 0.49986067102956
> Score (-lnL):             inf
> Lambda: 0.40553337462359
> ......
> Lambda: 0.5059341117188
> Score (-lnL):             inf
> 
> Families with largest size differentials:
> OG0000027: 23
> OG0002055: 17
> ......
> OG0000025: 9
> OG0000012: 9
> 
> You may want to try removing the top few families with the largest difference
> between the max and min counts and then re-run the analysis.
> 
> Failed to initialize any reasonable values

I tried to remove orthologous groups with differential >10. However, the error still exists. 

The gene count file was created using orthofinder. Do you have any idea how to deal with such problem? 

Thanks~
Xiaoyuan

**Comment by evolbeginner** [comment]
Are you analyzing gene families of bacterial genomes? I might be wrong but it seems to me that cafe may be more suitable for analyzing those with many duplicates, which may not be common for bacteria.

**Comment by Xiaojun928** [comment]
> Are you analyzing gene families of bacterial genomes? I might be wrong but it seems to me that cafe may be more suitable for analyzing those with many duplicates, which may not be common for bacteria.

Hi,

Thank you for your reply. I'm using CAFE for bacterial genomes. Maybe I have to choose other softwares.

Many thanks.
Xiaoyuan

**Comment by benfulton** [comment, marked as answer]
It's not _wrong_ to analyze bacteria, but it may not be as accurate. Dr. Hahn's main concern would be that lots of gene gain in bacteria is via horizontal gene transfer, not duplication - CAFE doesn't model HGT. Also, bacteria trees are difficult to work with because they can be extremely deep in time, and difficult to accurately date as the fossils all look the same.

**Comment by caord-1** [comment]
Hello, I also encountered this problem. I want to know how you have solved it. please!

**Comment by Wu-tz** [comment]
Hello，have you solved this problem? I used animal genomes but still no solved it when I removed families with largest size differentials.

**Comment by benfulton** [comment]
You might look at some of the suggestions in https://github.com/hahnlab/CAFE5/discussions/132
